build(bazel): some cleanup tasks

kormide opened the pull request:
@devversion @josephperrott @gregmagolan 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

Updated CLI sources version to match with `main`, removed some unused package.json scripts.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

kormide committed:
build(bazel): update CLI sources ref for aio build

It was out of date with the current legacy build

kormide committed:
build(bazel): remove unused AIO package.json scripts

No longer needed after the migration

kormide committed:
build(bazel): fix an AIO bug where symlinked assets were not loaded on
Windows

Added "followSymlinks" blocks on the assets.

kormide committed:
build(bazel): consolidate windows chromium path workaround

Add the hack to a central location so that it's easier to fix later.

kormide committed:
build(bazel): update rnj and remove launcher patch

rules_nodejs released  a fix for the RUNFILES issue.

kormide committed:
build(bazel): fix missing dgeni dependencies

Fixes the 'Export has no declarations: unknown' warning and
unresolved type warnings.

kormide committed:
build(bazel): fix a regression where github links were incorrectly
written

Incorrectly prepended 'packages/' to docs that were under aio/content.

kormide commented on aio/tools/transforms/angular-api-package/processors/fixupProjectRelativePath.js:
```diff
@@ -4,7 +4,7 @@ module.exports = function fixupProjectRelativePath(API_DOC_TYPES) {
     $process(docs) {
       docs.forEach(doc => {
-        if (API_DOC_TYPES.indexOf(doc.docType) !== -1 && doc.fileInfo && doc.fileInfo.projectRelativePath) {
+        if (API_DOC_TYPES.indexOf(doc.docType) !== -1 && doc.fileInfo && doc.fileInfo.realFilePath) {
```
This fixes a regression caused by https://github.com/angular/angular/commit/92a06f7981caf3685d3e1a97ddef4c87eac4de91#diff-077dd6145fe7c7d7e36918be05c1ccac4aa3393ff98f743feaa0730ea2d86b6aR7.

Basically, the `realX` properties are undefined when the doc is outside of `packages/`, e.g., a doc inside of `/aio/content`. `projectRelativePath` doesn't do any resolution so it's always defined, causing some files to get the `pacakges/` prefix that shouldn't.

kormide commented on aio/BUILD.bazel:
```diff
@@ -33,15 +33,22 @@ npm_package_bin(
     args = ["./$(BINDIR)/aio/tools/transforms/angular.io-package"],
-    data = ["//aio/tools/transforms/angular.io-package"],
+    data = [
+        "//aio/tools/transforms/angular.io-package",
+        # Dgeni typescript package crawls Angular dependencies
+        "@npm//:node_modules",
+    ],
     env = {
         "BAZEL_DGENI_OUTPUT_PATH": "$(@D)",
         "GIT_BIN": "$(GIT_BIN_PATH)",
     },
     exec_properties = ENABLE_NETWORK,
     output_dir = True,
     # Makes remote git calls to get previous versions
-    tags = ["requires-network"],
+    tags = [
+        "no-remote-exec",
+        "requires-network",
+    ],
```
Because dgeni needs to be able to walk angular's `node_modules` tree to resolve typescript types, I had to add `no-remote-exec` because we're going over the RBE limit. Limit is 70,000 and this puts it at ~96000.

devversion replied:
In the components repo we are just doing this explicitly. Can we do this here too? https://github.com/angular/components/blob/main/src/BUILD.bazel#L34-L39

devversion replied:
I believe you should be able to fine-grain select dependencies which are relevant for the TS API doc generation-like we do in the components repo.

kormide replied:
Good call. Turns out I only needed
```
        "@npm//rxjs",
        "@npm//@types/node",
```

kormide committed:
build(bazel): fix remote cache missses on Windows and OSX

Caused by chromium debug logs in @angular/build-tooling.

devversion commented on aio/angular.json:
```diff
@@ -49,10 +49,27 @@
-              "src/google385281288605d160.html",
+              // These asset blocks with "followSymlinks" are needed to properly
+              // serve assets on Windows due to running in a Bazel runfiles
+              // symlink forest.
```
Sorry, I feel like I keep asking on this symlinking stuff but I want to be sure we capture this in the relevant spots. Why do we need to follow symlinks for these files but not for the other generated files? Especially if we use `copy_to_bin`? Isn't everything a symlink in the runfiles directory?

This stuff can be captured in the commit messages too

kormide replied:
I'm not exactly sure why these assets were failing to load on Windows. @alan-agius4 pointed me to [this fix](https://github.com/angular/angular-cli/issues/20293#issuecomment-800962381) and it seemed to just work.

I want to double check why it's not needed for the other folders, or at least what makes the cases different. I'll do that and I'll try to write a better comment.

kormide replied:
I looked into this a bit further.

The assets are layed out in the symlink tree the way on all platforms: asset (source) files are symlinks, laid out in real folders.

Something different about the assets below these ones, `stackblitz/generated`, `example-zips/generates` etc. is that these are directory artifacts, so the folder itself is a symlink to the output tree but then all of the files are real. 

For whatever reason, architect on Windows has issues following symlinks to files but not symlinked directories to real files. I checked to see whether this was related to the `"serve"` target using assets from the source tree rather than the copied-to-bin version, but it's the same result.

So this may just be an architect issue (fyi @alan-agius4 ).

@devversion I'll improve the comment but are you okay with this workaround for now?

kormide committed:
build(bazel): fix regressions for docs-watch

Some earlier refactoring prevented the watchr script from running in
the source tree. There was also a Windows issue where running architect
didn't preserve symlinks causing the app to not be served.
